### ISSUE TITLE
java/overly-permissive-file-permission False Negatives

### ISSUE THREAD

**Issue by 9iang22** [returntocorp/sgrep-rules#3818]
## Description
The rule misses overly permissive file permissions when:
1. Using legacy `java.io.File` API instead of NIO
2. Using alternative NIO sink methods (`createTempFile`, `createFile`, `PosixFileAttributeView`)
3. Using symbolic mode strings or permissions set conditionally
## Case 1: Legacy java.io.File API
```java
import java.io.File;
public class FileApi {
    public static void notOk() {
        File file = new File("/var/opt/config.xml");
        // FN: setReadable/setWritable with ownerOnly=false grants to all users
        file.setReadable(true, false);
        file.setWritable(true, false);
        file.setExecutable(true, false);
    }
}
```
## Case 2: Alternative NIO Sink Methods
```java
import java.nio.file.Files;
import java.nio.file.Paths;
import java.nio.file.attribute.PosixFileAttributeView;
import java.nio.file.attribute.PosixFilePermissions;
public class FileApi {
    public static void notOk() throws Exception {
        // FN: PosixFileAttributeView.setPermissions not tracked
        PosixFileAttributeView attrs = Files.getFileAttributeView(
            Paths.get("/var/opt/config.xml"), PosixFileAttributeView.class);
        attrs.setPermissions(PosixFilePermissions.fromString("rw-rw-rw-"));
        // FN: Files.createTempDirectory not tracked
        Files.createTempDirectory("temp-", 
            PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("rw-rw-rw-")));
        // FN: Files.createTempFile not tracked
        Files.createTempFile(Paths.get("/tmp"), "prefix", "suffix",
            PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("rw-r--r--")));
        // FN: Files.createFile not tracked
        Files.createFile(Paths.get("/var/opt/config.xml"),
            PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("rw-rw-rw-")));
    }
}
```
## Case 3: Symbolic Mode Strings and Conditional Adds
```java
import java.nio.file.Files;
import java.nio.file.Paths;
import java.nio.file.attribute.PosixFilePermission;
import java.nio.file.attribute.PosixFilePermissions;
import java.util.HashSet;
import java.util.Set;
public class FileApi {
    public static void notOk() throws Exception {
        // FN: Symbolic mode "ugo+x" doesn't match 9-char regex
        Files.setPosixFilePermissions(Paths.get("/var/opt/config.xml"), 
            PosixFilePermissions.fromString("ugo+x"));
        // FN: Conditional add breaks pattern flow
        Set<PosixFilePermission> perms = new HashSet<>();
        if (true) perms.add(PosixFilePermission.OTHERS_WRITE);
        Files.setPosixFilePermissions(Paths.get("/var/opt/config.xml"), perms);
    }
}
```
